src/web/src/app/space/service.ts

```diff
@@ -53,4 +141,162 @@ export class SpaceService {
+			pipe(delay(2000));
+	}
+
+	getInvitationListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceInvitation>> {
+		let grouped: SpaceInvitation[] = [];
+		this.invitations.forEach(function(invitation) {
+			if (invitation.spaceId === spaceId){
+				grouped.push(invitation);
+			}
+		});
+		const res: PaginatorApi<SpaceInvitation> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res)
+			.pipe(delay(2000));
+	}
+
+	getLinkListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceLink>> {
+		let grouped: SpaceLink[] = [];
+		this.links.forEach(function(link) {
+			if (link.spaceId === spaceId){
+				grouped.push(link);
+			}
+		});
+		const res: PaginatorApi<SpaceLink> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res)
+			.pipe(delay(2000));
+	}
+
+	getUserListForSpace(spaceId: string, limit: number, page: number): Observable<PaginatorApi<SpaceUser>> {
+		let grouped: SpaceUser[] = [];
+		this.users.forEach(function(user) {
+			if (user.spaceId === spaceId){
+				grouped.push(user);
+			}
+		});
+		const res: PaginatorApi<SpaceUser> = {
+			count: grouped.length,
+			results: grouped.slice(limit * page, limit * page + limit),
+		};
+		return of(res)
+			.pipe(delay(2000));
+	}
+
+	getLinkList(limit: number, page: number): Observable<PaginatorApi<SpaceLink>> {
+		const res: PaginatorApi<SpaceLink> = {
+			count: this.links.length,
+			results: this.links.slice(limit * page, limit * page + limit),
+		};
+		return of(res)
+			.pipe(delay(2000));
+	}
+
+	isExists(key: string): Observable<boolean> {
+		return of(this.spaces.some(s => s.key === key))
+			.pipe(delay(2000));
+	}
+
+	createNew(name: string, key: string, requestsAllowed: boolean): Observable<boolean> {
+		this.spaces.push({
+			id: crypto.randomUUID(),
+			name: name,
+			key: key,
+			requestsAllowed: requestsAllowed,
+			createdAt: new Date()
+		});
+		return of(true)
+			.pipe(delay(2000));
+	}
+
+	createInvitation(spaceId: string, userId: string, role: string, creatorId: string): Observable<boolean> {
+		this.invitations.push({
+			id: Math.random(),
+			spaceId: spaceId,
+			userId: userId,
+			role: role,
+			creatorId: creatorId,
+			createdAt: new Date(),
+		});
+		return of(true)
+			.pipe(delay(2000));
+	}
+
+	createLink(spaceId: string, creatorId: string, name: string, expiredAt: Date): Observable<boolean> {
+		this.links.push({
+			id: crypto.randomUUID(),
+			spaceId: spaceId,
+			creatorId: creatorId,
+			name: name,
+			createdAt: new Date(),
+			expiredAt: expiredAt
+		});
+		return of(true)
+			.pipe(delay(2000));
+	}
+
+	delInvitationById(invitationId: number): Observable<boolean> {
+		const index = this.invitations.findIndex(x => x.id === invitationId);
+		if (index > -1) {
+			this.invitations.splice(index, 1);
+		}
+		return of(true).
+			pipe(delay(2000));
```

Comment:
А для чего такая проверка на -1? В каких ситуациях может удаляться элемент которого нет?
Давай добавим возврат ошибки и во всех таких функциях перенесём удаление после таймаута, чтобы отловить возможные проблемы с асинхронностью.
```suggestion
		return of(true).pipe(
			delay(2000),
			switchMap(val => {
				const index = this.invitations.findIndex(x => x.id === invitationId);
				if (index < 0)
				{
					const err = new Error("Invalid Id");
					return throwError(err);
				}
				this.invitations.splice(index, 1);
				return of(val);
			}),
			// TODO: requestWatcher call
		);
```

Reply:
Ок, на завтра запланировал таймаут перенести.

-1 взят с ответа на SO)

Reply:
Готово, -1 был взял с ответа в SO.